docs/book/src/developer/providers/v1alpha2-to-v1alpha3.md

```diff
@@ -153,7 +153,27 @@ outside of the existing module.
 
 - A new annotation `cluster.x-k8s.io/paused` provides the ability to pause reconciliation on specific objects.
 - A new field `Cluster.Spec.Paused` provides the ability to pause reconciliation on a Cluster and all associated objects.
-- A helper function `util.IsPaused` can be used on any Kubernetes object associated with a Cluster.
+- A helper function `util.IsPaused` can be used on any Kubernetes object associated with a Cluster and can be used during a Reconcile loop:
```

Comment:
In capa, we've had to wrap `util.IsPaused` to accept `nil` for the cluster resource, should this also be incorporated here? With this it is possible to also use `util.IsPaused` as a predicate on the controller to filter out updates for a resource when it has the paused annotation such as: https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/master/controllers/awscluster_controller.go#L200

Reply:
Maybe we could have a different function for when we want to look only at the paused annotation? What's the use case or scenario you're using IsPaused with nil?

Reply:
> Maybe we could have a different function for when we want to look only at the paused annotation?

Yes, that would definitely fit the use case we have as well.

> What's the use case or scenario you're using IsPaused with nil?

For one example, we are using it to set predicates on the watch for AWSCluster when we are building the controller in SetupWithManager, where we wouldn't have a cluster available.

Reply:
Added `util.HasPausedAnnotation`, lmk what you think

Reply:
looks great, thanks!